app/src/app/voting/voting.component.ts

```diff
@@ -67,10 +76,121 @@ export class VotingComponent implements OnInit {
     }
   }
 
+  composeAndSetData() {
+    combineLatest(
+      this.db.list("landsraad/votes/" + this.questionId).snapshotChanges(),
+      this.db.list("landsraad/answers/" + this.questionId).snapshotChanges(),
+      this.db.list("landsraad/votingRights").snapshotChanges(),
+      (votes, answers, votingRights) => {
+        return { 
+          votes: votes, answers: answers, votingRights: votingRights}
+      }
+    ).pipe(
+      tap(
+        combined => {
+          // voting rights ids to house names
+          let votingRightToHouseMap: GenericFirebaseSnapshotMapping = {}
+          combined.votingRights.forEach(votingRightSnap => {
+            // malorod is for testing, skipping for the general table
+            let houseName = votingRightSnap.payload.val()["name"].split(" ")[0]
+            if (houseName.toLowerCase() === "malorod") {
+              return
+            }
+            votingRightToHouseMap[votingRightSnap.key] = houseName
+          })
+
+          // current question answers ids to names
+          let currentQuestionAnswersMap: GenericFirebaseSnapshotMapping = {}
+          combined.answers.forEach((row) => {
+              currentQuestionAnswersMap[row.key] = row.payload.val()["name"]
+          })
+          // votes per house
+          let votesByHouse: VoteByHouse = {}
+          let finalData: TableRow[] = [...Object.values(currentQuestionAnswersMap).map(answer => ({dekret: answer}))]
+          combined.votes.forEach(voteSnap => {
+            let answer = voteSnap.payload.val()
+            let answerId = Object.keys(answer)[0]
+            let answerString: string = currentQuestionAnswersMap[answerId]
+            let votedBy: string = votingRightToHouseMap[voteSnap.key]
+            let votes = answer[answerId]
+            let answerIndexInData: number = finalData.findIndex(row => row.dekret === answerString)
+            // either create new entry in row or add votes to current
+            finalData[answerIndexInData].hasOwnProperty(votedBy) ? 
+              finalData[answerIndexInData][votedBy] = finalData[answerIndexInData][votedBy] + votes :
+                finalData[answerIndexInData][votedBy] = votes
+            // add total row
+            let voteInVotesByHouse = Object.keys(votesByHouse).includes(votedBy)
+            voteInVotesByHouse ?
+              votesByHouse[votedBy] = votesByHouse[votedBy] + votes :
+                votesByHouse[votedBy] = votes
+          })
+          // add total per answer
+          finalData = finalData.map((row: TableRow) => {
+            let total = Object.values(row)
+              .map(v => typeof(v) === "number" ? v : undefined)
+              .filter(v => v !== undefined)
```

Comment:
When does the undefined case happen?

Reply:
When we browse the table row by row, fist column is string because it is one of the answers to the current question. But I changed the `map` in the first row so that it returns `0` when the type of the cell is something else than number and then we do not need the `filter`. And the `0` also does not change the results, so it is fine to use it.